esp-wifi/build.rs

```diff
@@ -105,6 +105,7 @@ fn main() -> Result<(), Box<dyn Error>> {
             ("dynamic_rx_buf_num", Value::UnsignedInteger(32), "WiFi dynamic RX buffer number. See [ESP-IDF Programming Guide](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/network/esp_wifi.html#_CPPv418wifi_init_config_t)"),
             ("static_tx_buf_num", Value::UnsignedInteger(0), "WiFi static TX buffer number. See [ESP-IDF Programming Guide](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/network/esp_wifi.html#_CPPv418wifi_init_config_t)"),
             ("dynamic_tx_buf_num", Value::UnsignedInteger(32), "WiFi dynamic TX buffer number. See [ESP-IDF Programming Guide](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/network/esp_wifi.html#_CPPv418wifi_init_config_t)"),
+            ("csi_enable", Value::Bool(false), "WiFi channel state information enable flag. See [ESP-IDF Programming Guide](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/network/esp_wifi.html#_CPPv418wifi_init_config_t)"),
```

Comment:
Is this something a library (rather than an application) would want to enable? (I feel like the answer here is yes)
In which case this should be a feature rather than a config.

Reply:
Not sure - if a library expects it but it's not set it will not compile (which is not nice - I agree). On the other hand, if a library enables it but it's unused (e.g. hidden behind some other feature of that library) and it's not really used it still degrades performance or at least increases memory consumption

Reply:
The unused problem still exists if the library refuses to compile without the config enabled, as the user would be be forced to enable the csi feature, even if the library doesn't use it.